src/test/java/io/vertx/core/FutureTest.java

```diff
@@ -1570,4 +1573,38 @@ public void testFromCompletionStageWithContext() {
 
     await();
   }
+
+  @Test
+  public void testCompletedFuturesContext() {
+    waitFor(4);
+
+    Thread testThread = Thread.currentThread();
+    ContextInternal context = (ContextInternal) vertx.getOrCreateContext();
+
+    Future.succeededFuture().onSuccess(v -> {
+      assertSame(testThread, Thread.currentThread());
+      assertNull(Vertx.currentContext());
+      complete();
+    });
+
+    context.succeededFuture().onSuccess(v -> {
+      assertNotSame(testThread, Thread.currentThread());
```

Comment:
can you replace this check with event-loop thread instead of not being the junit thread ?

Reply:
Do you mean this?

```java
    context.succeededFuture().onSuccess(v -> {
      assertNotSame(testThread, Thread.currentThread());
      assertThat(Thread.currentThread(), is(instanceOf(VertxThread.class)));
      assertSame(context, Vertx.currentContext());
      complete();
    });
```

Reply:
or even better the thread of the context that is associated with the future (that you can latch before)

Reply:
I think something like:

```java
ctx.runOnContext(v -> {
  Thread ctxThread = Thread.currentThread();
  new Thread(() -> {
    ctx.succeededFuture().onSuccess(v -> {
      // Checks here
    });
  }).start();
});
```

Reply:
What do we want to verify here? That a standard context always operates on the same thread? I believe we could take this for granted here.

Reply:
this is what should be checked, given that this implementation is different from `FutureImpl`. I was taking for granted that this impl was bug free 2 days ago :-)

Reply:
@vietj I've added some checks. PTAL